app/lib/annual_report/percentiles.rb

```diff
@@ -1,62 +1,37 @@
 # frozen_string_literal: true
 
 class AnnualReport::Percentiles < AnnualReport::Source
+  def prepare
+    AnnualReport::StatusesPerAccountCount.connection.exec_query(<<~SQL.squish, nil, [year, year_as_snowflake_range.first, year_as_snowflake_range.last])
+      INSERT INTO annual_report_statuses_per_account_counts (year, account_id, statuses_count)
+      SELECT $1, account_id, count(*)
+      FROM statuses
+      WHERE id BETWEEN $2 AND $3
+      AND (local OR uri IS NULL)
+      GROUP BY account_id
+      ON CONFLICT (year, account_id) DO NOTHING
+    SQL
+  end
```

Comment:
Maybe I am missing something here, but this looks to me like it will prepare this for *all* accounts (which might be totally fine), but it is called for every account individually?

Reply:
The idea is that the `prepare` method would only be called once before a whole batch of annual reports need to be calculated. There is currently no code that actually generates a batch of annual reports, but presumably you'd do `AnnualReport.new(nil, 2023).prepare` and then queue up all the individual ones with `AnnualReport.new(account, 2023).generate`. Does that make sense?

Reply:
How about making `prepare` be a class method then and passing in the year as a parameter?

Reply:
Okay, done. Had to duplicate a bit of the `year_as_snowflake_range` code for it though.

Reply:
Nice. I really believe it makes more sense this way.